using-mobile-extensions/mobile-core/signals/signals-extension-and-rules-engine-integration.md

```diff
@@ -2,55 +2,90 @@
 
 ### Using tokens in Postbacks and PII rule actions <a id="using-tokens-in-postbacks-and-pii-rule-actions"></a>
 
-`Postbacks` and `PII` actions allow you to specify a `templateUrl` field and an optional `postbody`field that allow you to specify which tokens will be expanded by the Experience Platform SDKs when the postback or PII network call is triggered. For more information on tokens, see [Rule Tokens](https://aep-sdks.gitbook.io/docs/using-mobile-extensions/mobile-core/signals/signals-extension-and-rules-engine-integration#rules-tokens).
+`Send Postback` and `Send PII` actions allow you to specify a `URL` field and an optional `Post Body` field. You can specify which tokens should be expanded by the Experience Platform SDKs when the post back or PII network call is triggered. For more information on tokens, see [Rule Tokens](https://aep-sdks.gitbook.io/docs/using-mobile-extensions/mobile-core/signals/signals-extension-and-rules-engine-integration#rules-tokens).
 
-To use data that is passed to the `collectPii` API to form a token, the format is:
+**Example**
+
+Here is an example of how to use the data that is passed to the MobileCore (Android) / ACPCore (iOS) `collectPii` API to form a token:
+
+**1. In the mobile application, call collectPII to fire Event with context data**
 
 {% tabs %}
 {% tab title="Android" %}
-**Call collectPII to fire Event with context data**
 
 ```java
+Signal.registerExtension();
+...
 Map<String, String> data = new HashMap<String, String>();
-data.put("user_email", "user_001@google.com");
+data.put("user_email", "user_001@example.com");
 MobileCore.collectPII(data);
 ```
 
-**Use above contaxt data in `templateUrl`**
-
-```text
-https://my.company.com/users?email={%contextdata.user_email%}
-```
 {% endtab %}
 
 {% tab title="iOS" %}
 
+**Objective-C Example** 
+
+```objective-c
+[ACPSignal registerExtension];
+...
+[ACPCore collectPii:data:@{@"user_email" : @"user_001@example.com"}];
+```
+
+**Swift Example**
+
+```swift
+ACPSignal.registerExtension()
+...
+let piiContextData: [String: String] = ["user_email" : "user_001@example.com"]
+ACPCore.collectPii(piiContextData) 
+```
+
 {% endtab %}
 {% endtabs %}
 
-**Tip**: `mypii` is the key in the data dictionary that is passed to the `collectPii` API.
+**2. Using Adobe Launch create a Data element for the user_email context data key**
+
+![Data Element Example for Collect PII context data key](../../../.gitbook/assets/data_element_example_collect_pii.png)
+
+**3. Using Adobe Launch create a new rule for sending a post back**
+
+Create a new rule by selecting the Event Mobile Core Collect PII and the Action Mobile Core Send Postback as in the following images.
+
+![Rule example using Collect PII event and Postback action](../../../.gitbook/assets/postback_pii_token_example.png)
+
+**4. Use the above data element in the Postback action**
+
+Edit the `Send Postback` action and include the following URL in the corresponding edit box:
```

Comment:
This is a step, so please add a number.

Reply:
The step name is mentioned above,step #4